packages/aws-cdk-lib/aws-stepfunctions/lib/util.ts

```diff
@@ -1,4 +1,35 @@
+import { ActivityProps } from './activity';
+import { StateMachineProps } from './state-machine';
+import { IKey } from '../../aws-kms';
+import { Duration } from '../../core';
+
 export function noEmptyObject<A>(o: Record<string, A>): Record<string, A> | undefined {
   if (Object.keys(o).length === 0) { return undefined; }
   return o;
+}
+
+export function validateEncryptionConfiguration(kmsKey: IKey | undefined, kmsDataKeyReusePeriodSeconds: Duration | undefined) {
+  if (kmsKey === undefined && kmsDataKeyReusePeriodSeconds !== undefined) {
+    throw new Error('You cannot set kmsDataKeyReusePeriodSeconds without providing a kms key');
+  }
+
+  if (kmsKey !== undefined && kmsDataKeyReusePeriodSeconds !== undefined
+    && isInValidKmsDataKeyReusePeriodSeconds(kmsDataKeyReusePeriodSeconds)) {
+    throw new Error('kmsDataKeyReusePeriodSeconds needs to be a value between 60 and 900');
+  }
+}
+function isInValidKmsDataKeyReusePeriodSeconds(kmsDataKeyReusePeriodSeconds: Duration) {
+  return kmsDataKeyReusePeriodSeconds < Duration.seconds(60) || kmsDataKeyReusePeriodSeconds > Duration.seconds(900);
+}
+
+export function constructEncryptionConfiguration(props: StateMachineProps | ActivityProps, defaultPeriodSeconds: number) {
```

Comment:
- With the current [implementation](https://github.com/aws/aws-cdk/blob/20bf8e532acc597378fda2142c5550d57f011d7f/packages/aws-cdk-lib/aws-stepfunctions/lib/util.ts#L25-L35) of `constructEncryptionConfiguration` and [usage](https://github.com/aws/aws-cdk/blob/20bf8e532acc597378fda2142c5550d57f011d7f/packages/aws-cdk-lib/aws-stepfunctions/lib/state-machine.ts#L550) , it will break snapshot tests for customers who haven't explicitly defined encryption configuration - since it's setting encryptionConfiguration to `{
      type: 'AWS_OWNED_KEY'
    }` by default
- I believe we should allow customers to explicitly set `AWS_OWNED_KEY` as in [CFN](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-stepfunctions-statemachine-encryptionconfiguration.html)
- I figured the best way solve both problems would be to have `constructEncryptionConfiguration()` continue to accept props of type `StateMachineProps | ActivityProps` and return undefined if the customer doesn't provide a KMS key or set a *new* flag via props called `useDefaultEncryption?` of type `boolean`. Here is my proposed implementation:

```
export function constructEncryptionConfiguration(props: StateMachineProps | ActivityProps) {
  const DEFAULT_KMS_DATA_KEY_REUSE_PERIOD_SECONDS = 300;
  validateEncryptionConfiguration(props);
  if (props?.kmsKey) {
    return {
      kmsKeyId: props.kmsKey.keyArn,
      kmsDataKeyReusePeriodSeconds: props?.kmsDataKeyReusePeriodSeconds ?
        props.kmsDataKeyReusePeriodSeconds.toSeconds() : DEFAULT_KMS_DATA_KEY_REUSE_PERIOD_SECONDS,
      type: 'CUSTOMER_MANAGED_KMS_KEY',
    };
  } else if (props?.useDefaultEncryption) {
    return {
      type: 'AWS_OWNED_KEY',
    };
  } else {
    return undefined;
  }
}
```

We would then set the `encryptionConfiguration` like so in StateMachine and Activity
```
const resource = new CfnStateMachine(this, 'Resource', {
      ...
      encryptionConfiguration: constructEncryptionConfiguration(props),
     ...
    });
```

Reply:
> I believe we should allow customers to explicitly set AWS_OWNED_KEY as in [CFN](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-stepfunctions-statemachine-encryptionconfiguration.html)

`Type: AWS_OWNED_KEY` is equivalent to not providing setting `EncryptionConfiguration` (`undefined`). I don't see a real use case for explicitly setting it and do not recommend adding a additional `useDefaultEncryption ` prop just for that.

Reply:
If `kmsKey` is not set, returning `undefined` EncryptionConfiguration seems fine

Reply:
> I figured the best way solve both problems would be to have constructEncryptionConfiguration() continue to accept props of type StateMachineProps | ActivityProps 

Not really seeing how that helps solve the problem better than passing `kmsKey` and `kmsDataKeyReusePeriodSeconds` as separate parameters

Reply:
Ok makes sense, will reflect those changes, thanks.